### ISSUE TITLE
Trouble using new AWS SDKs

### ISSUE THREAD

**Issue by wiatrp** [OpenC3/openc3#136]
I'm trying to use an AWS SDK (ground station) and I think the way the S3 SDK is implemented is causing issues with that. I'm not a ruby expert so there may be an easier/better way to do this but it seems like it may be worth it to refactor how the S3 SDK is implemented for future use of AWS SDKs. I wanted to put the AWS GS code in the same location as the S3 code so it can be used by other OpenC3 APIs/services in the future.

I implemented the AWS GS library the same way the AWS S3 library was implemented

1. Create `awsgs.rb` in `lib/openc3/utilities` for the AWS GS API
    ```
    autoload(:Aws, 'openc3/utilities/awsgs_autoload.rb')

    class GSUtils
      def self.list_satellites()
        client = Aws::S3::Client.new 
        satellites = client.list_satellites({}).satellites
      end
    end
    ```
1. Create `awsgs_autoload.rb` in `lib/openc3/utilities` to load the library
    ```
    require 'aws-sdk-groundstation'
    
    Aws.config.update({
      credentials: Aws::Credentials.new(
        ENV['AWS_GS_ACCESS_KEY_ID'],
        ENV['AWS_GS_SECRET_ACCESS_KEY']
      ),
      region: ENV['AWS_GS_REGION']
    })
    ```

I ran into two issues from this

1. The autoload call will replace any already existing constants. In this case the AWS S3 library is loaded first, then the AWS GS library is loaded. When this happens I get errors that the AWS S3 library doesn't exist.
    ```
    NameError (uninitialized constant Aws::S3
          rubys3_client = Aws::S3::Client.new
                             ^^^^):
    openc3 (5.0.9.pre.beta0) lib/openc3/models/target_model.rb:90:in `all_modified'
    ```

    To get around this I replaced both autoload calls with regular require calls
    ```
    require 'openc3/utilities/s3_autoload.rb'
    require 'aws-sdk-groundstation'
    ```
1. The Aws config used for S3 (`s3_autoload.rb`) sets the global config for Aws which is used in every Aws call. This config is specific to S3 and breaks other AWS SDK calls. I had to modify it to look like below. This way, this specific config is only called for the S3 library
    ```
    require 'aws-sdk-s3'
    
    Aws.config.update(
      s3: {
        endpoint: ENV['OPENC3_S3_URL'] || (ENV['OPENC3_DEVEL'] ? 'http://127.0.0.1:9000' : 'http://openc3-minio:9000'),
        access_key_id: ENV['OPENC3_MINIO_USERNAME'],
        secret_access_key: ENV['OPENC3_MINIO_PASSWORD'],
        force_path_style: true,
        region: 'us-east-1'
      }
    )
    ```

Any thoughts on this?

**Comment by ryanmelt** [comment]
I like the config update change.  Will have to think about the autoload.  I'm pretty sure the autoload greatly improved startup time for processes that don't use S3.  I'd like to find a way to make that still work with other libraries.

**Comment by bradenvirdellgenevatech** [comment]
Hello, Just wondering if the ability to have multiple AWS configs for different AWS services will be added in an upcoming update? GTI is looking to use this ability in the near future.

**Comment by ryanmelt** [comment]
We will get a fix for this into 5.0.11
